### DISCUSSION TITLE
singularize different behaviour with 'active_support'

### DISCUSSION THREAD

**Discussion by pgorgita** [jeremyevans/sequel#2253] [General]
Hi Jeremy,

`#singularize` returns differently when `active_support/inflector` is required:

```ruby
# ruby-3.2.2, sequel-5.86.0, activesupport-7.2.2
require 'sequel '
include Sequel::Inflections
singularize "rfss"
=> "rfs"

require 'active_support/inflector'
singularize "rfss"
=> "rfss"
```

I see it should use pre-defined `#singularize`, when it responds:
https://github.com/jeremyevans/sequel/blob/6a7f9158adfc49d0099e7ef7b5a6a6bbb7c0ea4b/lib/sequel/model/inflections.rb#L136

It gets tricky with Associations.

I can override it in sequel or in 'active_support'. 
My question is, is there any other solution?
Do you plan to harmonize the singularize(s)?
Is there a way to solely use sequel's solution bypassing any external support?

Cheers,

Paulo

**Comment by jeremyevans** [comment]
If you require active_support's inflector (or any library adding the inflection methods to string), Sequel will use it, as you can see in your code example. 

You can also configure Sequel's inflector without using a string inflector library:

```ruby
Sequel::Inflections.uncountable "rfss"
```

No further changes to the default inflections are planned.

**Comment by pgorgita** [comment]
Problem remains as there is a totally different behavior between ''inflection"s:

```ruby
require 'sequel'
include Sequel::Inflections
singularize "rfss" => "rfs"
pluralize _        => "rfs"

Sequel::Inflections.uncountable "rfss"
singularize "rfss" => "rfss"
pluralize _        => "rfss"

require 'active_support/inflector'
singularize "rfss" => "rfss"
pluralize _        => "rfsses"
```

Anyway, okay... I just need to ensure that I keep these non-standard schemas at the forefront of my mind and take care of them.
